tests/test_replwrap.py

```diff
@@ -25,10 +27,24 @@ def test_bash(self):
         res = bash.run_command("time")
         assert 'real' in res, res
 
-        # PAGER should be set to cat, otherwise man hangs
-        res = bash.run_command('man sleep', timeout=2)
+    def test_pager_as_cat(self):
+        " PAGER is set to cat, to prevent timeout in ``man sleep``. "
+        bash = replwrap.bash()
+        res = bash.run_command('man sleep', timeout=5)
         assert 'SLEEP' in res, res
 
+    def test_long_running_multiline(self):
+        " ensure the default timeout is used for multi-line commands. "
+        bash = replwrap.bash()
+        res = bash.run_command("echo begin\r\nsleep 2\r\necho done")
+        self.assertEqual(res.strip().splitlines(), ['begin', 'done'])
+
+    def test_long_running_continuation(self):
+        " also ensure timeout when used within continuation prompts. "
+        bash = replwrap.bash()
+        res = bash.run_command("echo begin\\\n;sleep 2\r\necho done")
```

Comment:
I changed it, but then the tests failed, and I looked closer, and I think you did this deliberately after all. It's to generate a continuation prompt:

```
$ echo begin\
+ ;
$ sleep 2
$ echo done
```